### DISCUSSION TITLE
Question about storage_discharge_limit property.

### DISCUSSION THREAD

**Discussion by Noodleyman** [WillCodeForCats/solaredge-modbus-multi#632] [General]
I am checking into an issue related to PredBat and Modbus Multi, and seeking some clarification with regards to the property **number.solaredge_i1_storage_discharge_limit**

When PredBat does a hold, it may set this property to a value of 0.

However, if I manually, or something automated sets this property to a value of 0 then it auto-resets back to 5000. The same does  not happen with a value of 1. 

Is a value of 0 invalid?

Thank you,

**Comment by deqw** [comment]
In what time do you see it auto-reset to 5000?

I looked into my code where I set the discharge limit and I made a comment there about a discharge limit reset to 5000 that occurs every day. But it can be set it to 0 anyway. I'm aware that this does not explain your persisting setting with a value of 1.

**Reply by Ashpork** [reply]
exactly, how quick does it go back to 5000, Mine will do this if I have an issue communicating with the inverter, in that it actually never actually took the requested value.

In my usage of PredBat, all the "Hold charge" and "freeze" modes, in my view aren't fit for purpose on a SE based system, so I have them all turned off, meaning it'll only do a discharge, charge and maximum self consumption, at the full 5000 available to me.

**Reply by deqw** [reply]
In my situation discharge limit resets at least once a day because the inverter does a reset by itself somewhere just before midnight where the inverter goes from I_STATUS_MPPT to I_STATUS_FAULT and back to I_STATUS_MPPT in about 10 seconds time.

This daily inverter reset was confirmed as normal behaviour by the installer and SolarEdge without any further explanation why. Of course wondering if others see this too.

**Reply by Ashpork** [reply]
I had a quick check for you. 

Mine routinely goes "unavailable" for 40-50 seconds at around 23:15 each night. I don't see the FAULT status during that time.

![image](https://github.com/user-attachments/assets/069f5048-36f8-4146-8738-ea16504658b2)

**Reply by deqw** [reply]
Thanks for confirming!

**Reply by deqw** [reply]
Mine is not "unavailable" at that moment so the fault is registered. It just takes ten seconds.
![Screenshot 2024-07-24 at 16 41 45](https://github.com/user-attachments/assets/c6715988-d03c-4c3e-92ed-61e24dbf057b)

**Comment by Noodleyman** [comment]
I'll do some extra checking, thank you for the feedback, it gives me something to test and validate further. I'll setup some test scenarios and report back. I did have an automation in place to listen for a 0 request and set t 1 instead, which seemed to work. I'll disable and experiment. and report back.

**Comment by Noodleyman** [comment]
I can see that I get the same kind of thing overnight, "became unavailable" around 23:02. 

I've not been able to replicate this today on the fly (I wish I had tested before starting this thread)... but I suspect what was said here may be correct. perhaps for some reason the request didn't get accpeted or received and it resets back to the previous value. I'm going to need to do more testing. 

Ultimately, it seems I got the answer to the question raised though. 0 is valid. :) 

Thanks all for taking the time to reply, I appreciate it.

**Reply by deqw** [reply]
My routine is just setting it back to the value I want when it is different from my discharge setting (using a Helper for that setting). I don't use PredBat btw. To confirm the overnight reset just set to 4500 or so and then check next morning (or go to bed late ;o).
